### PR TITLE
Kdesktop 1388 memory leaks step 2

### PR DESCRIPTION
**Pull request by ChristopheLarchier**

### CODE REVIEW

File: src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp [hunk 1]
```diff
@@ -504,25 +504,25 @@ void Snapshot::setValid(bool newIsValid) {
...
         }
 
-        auto result = names.insert(_items[*childId].name());
+        auto result = names.insert(_items.at(*childId).name());
         if (!result.second) {
             LOGW_WARN(Log::instance()->getLogger(), L"Snapshot integrity check failed, the folder named: \""
                                                             << SyncName2WStr(parentItem.name()).c_str() << L"\"("
                                                             << Utility::s2ws(parentItem.id()).c_str() << L") contains: \""
-                                                            << SyncName2WStr(_items[*childId].name()).c_str()
+                                                            << SyncName2WStr(_items.at(*childId).name()).c_str()
```
[the remark this answers is not in the document]
Comment (ChristopheLarchier) [reply src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp]: Done

File: src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp [hunk 2]
```diff
@@ -504,25 +504,25 @@ void Snapshot::setValid(bool newIsValid) {
...
                                                             << L"\" twice with two differents NodeId");
```
[the remark this answers is not in the document]
Comment (ChristopheLarchier) [reply src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp]: Done

File: src/libcommonserver/utility/utility.cpp [hunk 3]
```diff
@@ -913,4 +913,9 @@ std::string Utility::userName() {
     return userName_private();
 }
 
+ExitInfo Utility::terminateThreadFunction(ExitInfo exitInfo) {
```
Comment (ClementKunz) [inline src/libcommonserver/utility/utility.cpp:916]: So this method can be `void` now?

**Review by luc-guyot-infomaniak — COMMENTED** [review]
There is a significant overlap with the first PR (https://github.com/Infomaniak/desktop-kDrive/pull/394#pullrequestreview-2471944058). So I'll wait the first PR is merged to resume the review of this PR.

File: src/libcommonserver/utility/utility.cpp (hunk 3 above)
Comment (ChristopheLarchier) [reply src/libcommonserver/utility/utility.cpp]: Doesn't exist anymore after the merge of "Kdesktop 1313 memory leaks"

**Review by ClementKunz — APPROVED** [review]

**Pull request merged**